src/App.vue

```diff
@@ -489,78 +491,49 @@ export default {
-     * @param {Object} workspaceSettings - workspace's version of settings
-     * @returns {Object} An object with user's and workspace's settings merged (with workspace's settings taking priority)
+     * Update the settings stored in the store and on the server as well
+     * @param {Object} updatedSettings - details about the leaf nodes that the user has updated
      */
-    mergeSettings(userSettings, workspaceSettings) {
-      // making a deep clone of global default settings.
-      // the keys/values will be removed/updated according to user/workspace settings as we iterate
-      let mergedSettings = clonedeep(globalDefaultSettings);
-      for (let [headerName, headerDetails] of mergedSettings) {
-        // iterating on headers
-        let workspaceHeaders = [...workspaceSettings.keys()];
-        if (!workspaceHeaders.includes(headerName)) {
-          // if the current header name is not present in workspace settings,
-          // pick the details from user's settings and put it into merged settings object
-          mergedSettings.set(headerName, userSettings.get(headerName));
-          continue;
-        }
-        // if the current header name is present in workspace settings, use its scope information
-        mergedSettings.get(headerName).scope = workspaceSettings.get(headerName).scope;
+    updateSettings(updatedSettings) {
+      // The updatedSettings object contains the following keys
+      // headerName - name of the header to which the updated setting belongs to
+      // tabName - name of the tab to which the updated setting belongs to
+      // leafName - name of the updated leaf node
+      // newValue - the updated value
+      // isWorkspaceSetting - whether the updated setting is a workspace setting
+      let newWorkspaceSettings = null;
+      let newUserSettings = null;
+      Object.keys(updatedSettings).forEach((key) => {
+        let setting = updatedSettings[key];
 
-        for (let [tabName, tabDetails] of headerDetails.children) {
-          // iterating on tabs inside headerName
-          let workspaceTabs = [...workspaceSettings.get(headerName).children.keys()];
-          if (!workspaceTabs.includes(tabName)) {
-            // if the current tab name is not present in workspace settings,
-            // pick the details from user's settings and put it into merged settings object
-            mergedSettings
-              .get(headerName)
-              .children.set(tabName, userSettings.get(headerName).children.get(tabName));
-            continue;
-          }
-          // if the current tab name IS present in workspace settings, use its scope information
-          mergedSettings
-            .get(headerName)
-            .children.get(tabName).scope = workspaceSettings
-            .get(headerName)
-            .children.get(tabName).scope;
+        if (setting.isWorkspaceSetting) {
+          newWorkspaceSettings = clonedeep(this.activeWorkspaceSettings);
```

Comment:
this should be done only if `newWorkspaceSettings` is already null. For example, if two setting keys are updated and both of them are newWorkspaceSettings, in both the cases we'll fetch the activeWorkspaceSettings and change just one value right? So, only the last workspace setting change will remain and not all the changes, right?